src/GraphControl/Control/Grapher.cpp

```diff
@@ -66,6 +83,34 @@ namespace GraphControl
         }
     }
 
+    void Grapher::ScaleRange(double centerX, double centerY, double scale)
+    {
+        if (m_graph)
+        {
+            if (auto renderer = m_graph->GetRenderer())
+            {
+                if (SUCCEEDED(renderer->ScaleRange(centerX, centerY, scale)))
+                {
+                    m_renderMain->RunRenderPass();
```

Comment:
In other places you are checking if m_renderMain is null, should this happen here too?